File: app/src/main/java/com/baidu/tv/player/utils/QRCodeUtils.java
package com.baidu.tv.player.utils;

import android.graphics.Bitmap;
import android.graphics.Color;

import com.google.zxing.BarcodeFormat;
import com.google.zxing.EncodeHintType;
import com.google.zxing.WriterException;
import com.google.zxing.common.BitMatrix;
import com.google.zxing.qrcode.QRCodeWriter;

import java.util.HashMap;
import java.util.Map;

/**
 * 二维码工具类
 */
public class QRCodeUtils {
    
    /**
     * 生成二维码Bitmap
     * @param content 内容
     * @param width 宽度
     * @param height 高度
     * @return Bitmap
     */
    public static Bitmap createQRCodeBitmap(String content, int width, int height) {
        QRCodeWriter writer = new QRCodeWriter();
        try {
            // 设置编码提示，减少边距
            Map<EncodeHintType, Object> hints = new HashMap<>();
            hints.put(EncodeHintType.MARGIN, 0); // 设置边距为0
            
            BitMatrix bitMatrix = writer.encode(content, BarcodeFormat.QR_CODE, width, height, hints);
            int[] pixels = new int[width * height];
            for (int y = 0; y < height; y++) {
                for (int x = 0; x < width; x++) {
                    if (bitMatrix.get(x, y)) {
                        pixels[y * width + x] = Color.BLACK;
                    } else {
                        pixels[y * width + x] = Color.WHITE;
                    }
                }
            }
            Bitmap bitmap = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888);
            bitmap.setPixels(pixels, 0, width, 0, 0, width, height);
            return bitmap;
        } catch (WriterException e) {
            e.printStackTrace();
            return null;
        }
    }
}

File: app/src/main/java/com/baidu/tv/player/kt/util/QRCodeUtils.kt
package com.baidu.tv.player.kt.util

import android.graphics.Bitmap
import android.graphics.Color
import com.google.zxing.BarcodeFormat
import com.google.zxing.EncodeHintType
import com.google.zxing.qrcode.QRCodeWriter

/**
 * 二维码工具类。
 */
object QRCodeUtils {

    /**
     * 生成二维码 Bitmap。失败返回 null。
     * @param content 内容
     * @param width 宽度
     * @param height 高度
     */
    fun createQRCodeBitmap(content: String, width: Int, height: Int): Bitmap? = try {
        val writer = QRCodeWriter()
        val hints = mapOf(EncodeHintType.MARGIN to 0) // 边距设为 0
        val bitMatrix = writer.encode(content, BarcodeFormat.QR_CODE, width, height, hints)
        val pixels = IntArray(width * height) { idx ->
            val x = idx % width
            val y = idx / width
            if (bitMatrix.get(x, y)) Color.BLACK else Color.WHITE
        }
        Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888).apply {
            setPixels(pixels, 0, width, 0, 0, width, height)
        }
    } catch (e: Exception) {
        e.printStackTrace()
        null
    }
}
